server/controllers/petProfile.js

```diff
@@ -0,0 +1,57 @@
+const PetProfile = require("../models/PetProfile");
+const asyncHandler = require("express-async-handler");
+
+exports.createPetProfile = asyncHandler(async (req, res, next) => {
+  const { petName, ownerID, petType, petAge, petStatus } = req.body;
+
+
+
+  const petProfile = await PetProfile.create({
+    petName,
+    ownerID,
+    petType,
+    petAge,
+    petStatus,
+  });
+
+  res.status(200).json({ petProfile });
+});
+
+exports.updatePetProfile = asyncHandler(async (req, res, next) => {
+  const values = ({ _id, petName, ownerID, petType, petAge, petStatus } =
+    req.body);
+
+  for (let prop in values) {
+    if (!values[prop]) {
+      delete values[prop];  //it will remove fields who are undefined or null
```

Comment:
You don't need this comment as this line is descriptive.

Reply:
OK